src/cpl/mct/lnd_import_export.F90

```diff
@@ -215,8 +215,9 @@ real(r8) :: tdc, t               ! Kelvins to Celcius function and its input
        end if
        if ( (atm2lnd_inst%forc_solad_grc(g,1) < 0.0_r8) .or.  (atm2lnd_inst%forc_solad_grc(g,2) < 0.0_r8) &
        .or. (atm2lnd_inst%forc_solai_grc(g,1) < 0.0_r8) .or.  (atm2lnd_inst%forc_solai_grc(g,2) < 0.0_r8) ) then
-          call endrun( sub//' ERROR: One of the solar fields (indirect/diffuse, vis or near-IR)'// &
-                       ' from the atmosphere model is negative or zero' )
+!tcx temporary
```

Comment:
Why was this check comments out?

Reply:
It's a good question and something we should look into further.  I think the model was aborting on this error, so I thought I'd comment it out to see what was happening.  And it worked.  I guess I never circled back.  It could be that the solar forcing terms are slightly negative (like precip) or there could be some other issues.  We should look into this further.  I'm happy to put the endrun call back in and run some tests.  How should we proceed on this point?

Reply:
I think it would be good to test this in some manner.  First identifying which component of the SW/LW is causing the problem, then we can start to dig around more. Could we check with the WRF-CTSM LILAC effort to see if they have a check like this?

Reply:
I ran some tests and I just updated the implementation.  Like the precip, it looks like there are tiny negative values sneaking into the coupling.  It could be roundoff or it could be something like what we've encountered with the precip coupling from WRF.  At any rate, I have added a check (like precip) where if the solar is less than 0 but greater than -1.0e-16, then the solar is zeroed out.  Otherwise, the abort is still triggered for negative solar.  I think we may need to look further if/why WRF is producing tiny negative values for downward solar terms, but in the short term, I think this fix is reasonable.

Reply:
My latest commit which reactivates the endrun call for negative solar and sets tiny negative solar to zero can be seen here,

https://github.com/nna-rivers/CTSM/pull/2/commits/348b745b29eb3a015f05021655826f3bb5f284f6

Reply:
This fix sounds reasonable.  I will check into this with the WRF developers and look/ask around.  I've never noticed this before, but I've never looked for it before in WRF.